src/lib.rs

```diff
@@ -343,28 +343,6 @@ pub mod class {
         pub use crate::pyclass::CompareOp;
     }
 
-    /// Old module which contained some implementation details of the `#[pyproto]` module.
-    ///
-    /// Prefer using the same content from `pyo3::pyclass`, e.g. `use pyo3::pyclass::IterANextOutput` instead
-    /// of `use pyo3::class::pyasync::IterANextOutput`.
-    ///
-    /// For compatibility reasons this has not yet been removed, however will be done so
-    /// once <https://github.com/rust-lang/rust/issues/30827> is resolved.
-    pub mod pyasync {
-        pub use crate::pyclass::{IterANextOutput, PyIterANextOutput};
-    }
-
-    /// Old module which contained some implementation details of the `#[pyproto]` module.
-    ///
-    /// Prefer using the same content from `pyo3::pyclass`, e.g. `use pyo3::pyclass::IterNextOutput` instead
-    /// of `use pyo3::class::pyasync::IterNextOutput`.
-    ///
-    /// For compatibility reasons this has not yet been removed, however will be done so
-    /// once <https://github.com/rust-lang/rust/issues/30827> is resolved.
-    pub mod iter {
-        pub use crate::pyclass::{IterNextOutput, PyIterNextOutput};
-    }
```

Comment:
Yeah, I guess we should. I just like deleting code...

Reply:
Ok, they are deprecated now and the `IntoPyCallbackOutput` impl is reduced to the one case of directly going to `*mut ffi::PyObject` which means there does not need to be a specialization for this type at all. (Which makes sense, this is a dedicated type after all and we can just implement the trait directly. Only because `__(a)next__` need special handling for `Option` that is distinct from the already existing trait implementation do we need to go for specialization.)